### PR TITLE
MueLu: update blocked coord transfer fact

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu 

## Motivation
BlockedCoordinatesTransferFactory was not properly setup when called from an xml input deck. Also updated to work for multiple dofs per node (per block). This now allows for blocked operators to use coordinate dependent factories, such as zoltan2 rebalancing.

This is related to work being done for MHD using Drekar to allow for rebalancing of the operators on the coarse levels:
https://github.com/rppawlo/DrekarBase/issues/546

## Testing
To be added.

### CODE REVIEW

File: packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp [hunk 1]
```diff
@@ -109,14 +109,51 @@ namespace MueLu {
...
+      LO blkSize = 0;
+      LO otherBlkSize;
+      for(size_t i=0; i<numSubFactories; i++) {
+        if (rcp_dynamic_cast<const StridedMap>(coarseMap->getMap(i, thyraMode)) != Teuchos::null){
+          otherBlkSize = rcp_dynamic_cast<const StridedMap>(coarseMap->getMap(i, thyraMode))->getFixedBlockSize();
+        } else {
+          otherBlkSize = 1;
+        }
+        if( i>0 && blkSize/i != otherBlkSize)
+          GetOStream(Warnings1) << "BlockedCoordinatesTransferFactory: Subblocks have different stride. Might cause problems..." << std::endl;
```
Comment (jhux2) [inline packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp:136]: Should this be an error instead of a warning?

File: packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp [hunk 2]
```diff
@@ -109,14 +109,51 @@ namespace MueLu {
...
+        blkSize += otherBlkSize;
```
Comment (jhux2) [inline packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp:137]: Why is this added?

**Review by jhux2 — COMMENTED** [review]
Could you please add something in the motivation section, so that future us will know this is related to your MHD work?

File: packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp (hunk 1 above)
Comment (pohm01) [reply packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp]: Was originally thinking of allowing block sizes to be different. I have changed it to be an error to avoid unexpected behavior.

File: packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp (hunk 2 above)
Comment (pohm01) [reply packages/muelu/src/Misc/MueLu_BlockedCoordinatesTransferFactory_def.hpp]: I was playing around with allowing different block sizes. Couldn't quite figure it out, so now block sizes need to be the same between different larger blocks.

File: packages/muelu/test/unit_tests/BlockedRepartition.cpp [hunk 3]
```diff
@@ -938,14 +941,331 @@ namespace MueLuTests {
...
+
+  /**********************************************************************************/
+  /* CREATE INITIAL MATRIX                                                          */
+  /**********************************************************************************/
+  RCP<const Map> bigMap;
+  RCP<const Map> map1;
+  RCP<const Map> map2;
+  GO numElements = 400;
+  GO numElements1 = 200;
+  GO numElements2 = 200;
```
Comment (jhux2) [inline packages/muelu/test/unit_tests/BlockedRepartition.cpp:966]: @pohm01 Are the blocks allowed to have different numbers of elements?  If so, would the test be more general and/or exercise more parts of the blocked transfer factory?  It's fine if the answer is "no", I'm just wondering.

**Review by jhux2 — DISMISSED** [review]
Lgtm -- had one question, see my comment.

File: packages/muelu/test/unit_tests/BlockedRepartition.cpp (hunk 3 above)
Comment (pohm01) [reply packages/muelu/test/unit_tests/BlockedRepartition.cpp:966]: Currently there is only support for one set of coordinates that is shared between the blocks, so the blocks need the same number of elements and co-located dofs/nodes.

**Review by jhux2 — APPROVED** [review]
lgtm
